src/redux/books/books.js

```diff
@@ -0,0 +1,22 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = { books: [] };
+
+const booksSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    addBook(state, action) {
+      state.push(action.payload);
+    },
+
+    removeBook(state, action) {
+      const foundId = state.books.findIndex((book) => book.id === action.payload);
+      state.books.splice(foundId, 1);
+    },
+  },
+});
+
+export const booksActions = booksSlice.actions;
+
+export default booksSlice.reducer;
```

Comment:
- [ ] I can see that you used `createSlice` from `toolkit` to write your book's actions and reducers and this is not wrong, But according to the requirements, you should write your book's actions and reducer using the [ducks pattern](https://github.com/erikras/ducks-modular-redux). So kindly recreate this file using [ducks pattern](https://github.com/erikras/ducks-modular-redux) to follow the requirements 💯👌

> It should be something like

```
// Define Action Type
const ADD = 'bookstore/books/ADD';
const REMOVE = 'bookstore/books/REMOVE';

export default function booksReducer(state = [], action) {
  switch (action.type) {
    case ADD: return ... ;
    case REMOVE: return ... ;
    default: return state;
  }
}

export function addBook(book) { ... }

export function removeBook(id) { ... }
```

![-2](https://user-images.githubusercontent.com/45231276/184617059-a7187888-29c6-4ba1-8d71-447bc33cd965.PNG)